src/thermo/include/antioch/ideal_gas_internal_thermo.h

```diff
@@ -0,0 +1,147 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// Antioch - A Gas Dynamics Thermochemistry Library
+//
+// Copyright (C) 2014 Paul T. Bauman, Benjamin S. Kirk, Sylvain Plessis,
+//                    Roy H. Stonger
+// Copyright (C) 2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef ANTIOCH_IDEAL_GAS_INTERNAL_THERMO_H
+#define ANTIOCH_IDEAL_GAS_INTERNAL_THERMO_H
+
+namespace Antioch
+{
+
+  template <typename CoeffType>
+  class ChemicalMixture;
+
+  template <typename CoeffType>
+  class TempCache;
+
+
+  template <typename ExternalThermo, typename CoeffType = double>
```

Comment:
To obtain Cv_vib, when you don't have access to the vibrational levels, you use the relationship between Cv and Cp, you get Cv, then you substract Cv_tr, and there you have Cv_vib.

MacroThermo provides Cp.